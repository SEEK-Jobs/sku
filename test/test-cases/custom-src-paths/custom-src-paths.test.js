const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const waitForUrls = require('../../utils/waitForUrls');
const getAppSnapshot = require('../../utils/getAppSnapshot');
const startAssetServer = require('../../utils/assetServer');
const skuConfig = require('./sku.config');

const targetDirectory = `${__dirname}/dist`;

describe('custom-src-paths', () => {
  beforeAll(async () => {
    await runSkuScriptInDir('configure', __dirname);
  });

  describe('start', () => {
    const devServerUrl = `http://localhost:${skuConfig.port}`;
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('start', __dirname);
      await waitForUrls(devServerUrl);
    });

    afterAll(() => {
      server.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(devServerUrl);
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('build', () => {
    let closeAssetServer;

    beforeAll(async () => {
      await runSkuScriptInDir('build', __dirname);
      closeAssetServer = startAssetServer(4002, targetDirectory);
      await waitForUrls('http://localhost:4002');
    });

    afterAll(() => {
      closeAssetServer();
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(targetDirectory);
      expect(files).toMatchSnapshot();
    });

    it('should create valid app', async () => {
      const app = await getAppSnapshot('http://localhost:4002');
      expect(app).toMatchSnapshot();
    });
  });

  describe('format', () => {
    it('should format successfully', async () => {
      const {
        childProcess: { exitCode }
      } = await runSkuScriptInDir('format', __dirname);
      expect(exitCode).toEqual(0);
    });
  });

  describe('lint', () => {
    it('should lint successfully', async () => {
      const {
        childProcess: { exitCode }
      } = await runSkuScriptInDir('lint', __dirname);
      expect(exitCode).toEqual(0);
    });
  });
});
