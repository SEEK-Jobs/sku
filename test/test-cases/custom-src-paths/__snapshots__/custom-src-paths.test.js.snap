// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-src-paths build should create valid app 1`] = `
Object {
  "content": "<!DOCTYPE html><html><head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\"><div data-reactroot=\\"\\">Hello custom src paths</div></div>
    <script type=\\"text/javascript\\" src=\\"http://localhost:4002/runtime-2844b96e4645a1349e71.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4002/vendors~main-c6807d2c4e89eaba157e.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4002/main-a79d6fc51aec0310c471.js\\" crossorigin=\\"anonymous\\"></script>
  
</body></html>",
  "errors": Array [],
  "sourceHtml": "<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\"><div data-reactroot=\\"\\">Hello custom src paths</div></div>
    <script type=\\"text/javascript\\" src=\\"http://localhost:4002/runtime-2844b96e4645a1349e71.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4002/vendors~main-c6807d2c4e89eaba157e.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4002/main-a79d6fc51aec0310c471.js\\" crossorigin=\\"anonymous\\"></script>
  </body>
</html>",
  "warnings": Array [],
}
`;

exports[`custom-src-paths build should generate the expected files 1`] = `
Object {
  "index.html": "<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\"><div data-reactroot=\\"\\">Hello custom src paths</div></div>
    <script type=\\"text/javascript\\" src=\\"http://localhost:4002/runtime-2844b96e4645a1349e71.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4002/vendors~main-c6807d2c4e89eaba157e.js\\" crossorigin=\\"anonymous\\"></script>
<script type=\\"text/javascript\\" src=\\"http://localhost:4002/main-a79d6fc51aec0310c471.js\\" crossorigin=\\"anonymous\\"></script>
  </body>
</html>",
  "main-a79d6fc51aec0310c471.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-2844b96e4645a1349e71.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-c6807d2c4e89eaba157e.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`custom-src-paths start should start a development server 1`] = `
Object {
  "content": "<!DOCTYPE html><html><head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\"><div data-reactroot=\\"\\">Hello custom src paths</div></div>
    <script type=\\"text/javascript\\" src=\\"/runtime.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/vendors~main.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
  
</body></html>",
  "errors": Array [],
  "sourceHtml": "<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>hello-world</title>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    
  </head>
  <body>
    <div id=\\"app\\"><div data-reactroot=\\"\\">Hello custom src paths</div></div>
    <script type=\\"text/javascript\\" src=\\"/runtime.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/vendors~main.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
  </body>
</html>",
  "warnings": Array [],
}
`;
